lib/chef/provider/cron.rb

```diff
@@ -41,11 +41,34 @@ def initialize(new_resource, run_context)
       end
       attr_accessor :cron_exists, :cron_empty
 
+      def check_resource_semantics!
+        return if Process.euid == 0
+
+        etype = Chef::Exceptions::Cron
+        begin
+          nuid = Etc.getpwnam(new_resource.user).uid
+        rescue ArgumentError
+          emessage = "Error in #{new_resource.name}, #{new_resource.user} does not exists on this system."
+          events.provider_requirement_failed(new_resource.action, new_resource.name, etype, emessage)
+          if why_run?
+            events.whyrun_assumption(new_resource.action, new_resource.name, "Assuming user #{new_resource.user} would have been acted upon.")
+          else
+            raise etype, emessage
+          end
+        end
+        return if nuid == Process.euid
+
+        emessage = "Error in #{new_resource.name}, Chef can't modify another user crontab when not running as root."
+        events.provider_requirement_failed(new_resource.action, new_resource.name, etype, emessage)
+        raise etype, emessage
+      end
```

Comment:
this should probably be moved to `defined_resource_requirements` and use the typical assertion syntax there.  if the point is to protect load_current_resource from exploding, then LCR really needs to be made 'safe' for the cases when the user doesn't exist and/or the thing isn't being run as root (should just return a bunch of nil stuff for what it can't get at).

what you're doing here with directly handling your own why run assertions is waaaaaay too complicated to follow for 99% of humans.

Reply:
Well, it's starting to be quite old :) I think this was made on 13 codebase at first.

I'll see to understand the define_resource_requirement then, any hint to bootstrap me?

Reply:
BTW, the goal is to fail the run with an informative message if the user doesn't exist as the resource can't do it's job.

Reply:
here's an example of trying to modify a group that doesn't exist, which is the closest i could find offhand:

https://github.com/chef/chef/blob/master/lib/chef/provider/group.rb#L54-L59